### PR TITLE
Replace VerbTypes with verb classes

### PR DESCRIPTION
**Pull request by ElectroJr**
This PR removes the `VerbType` enum flags in favour of just having different verb types be classes that inherit from the `Verb` class. So `GetAlternativeVerbsEvent` and similar events are now replaced with `GetVerbsEvent<AlternativeVerb>`. 

The main motivation for doing this is so that if verbs are used outside of the context menu in more specialized roles, those verbs can store information specific to that verb type. E.g., #6489 adds an examine verb type that appear in the examine popup. Having an ExamineVerb subclass would allow those verbs to have data specific to the pop-up & examining, rather than only generic information for the context menu. In particular, the question came up of how to best prevent a specific examine verbs from appearing in the examine pop-up without doing awkward manual filtering.

This PR results in a bunch of small changes to miscellaneous files due to the get-verb-event changes. The main files with changes include:
- `Verb.cs`
- `VerbEvents.cs`
- serve/client/shared `VerbSystem`
- List & Invoke verb commands
- `VerbMenuPresenter.cs`
- `VerbMenuElement.cs`

### CODE REVIEW

File: Content.Shared/Verbs/Verb.cs [hunk 1]
```diff
@@ -180,5 +187,80 @@ public int CompareTo(object? obj)
...
+        ///     Though probably this should use reflection or something.
+        /// </remarks>
+        public static Dictionary<Type, string> VerbTypes = new()
+        {
+            { typeof(Verb), "Verb" },
+            { typeof(InteractionVerb), "InteractionVerb" },
+            { typeof(AlternativeVerb), "AlternativeVerb" },
+            { typeof(ActivationVerb), "ActivationVerb" },
+        };
+    }
```
Comment (ElectroJr) [inline Content.Shared/Verbs/Verb.cs:205]: This is probably the jankiest part of this. Apparently `Type` isn't serializable, so AFAIK the easiest way of communicating to the server what types of verbs have been requested is to just turn it into a list of strings and then back again. Though the keys of this dictionary are also useful when enumerating over all verb types,

Comment (mirrorcult) [reply Content.Shared/Verbs/Verb.cs:205]: cant you just store a list of verbtypes and then use `nameof()`

File: Content.Shared/Interaction/SharedInteractionSystem.cs [hunk 2]
```diff
@@ -736,7 +736,7 @@ protected virtual void BeginDelay(UseDelayComponent? component = null)
         public bool AltInteract(EntityUid user, EntityUid target)
         {
             // Get list of alt-interact verbs
-            var verbs = _verbSystem.GetLocalVerbs(target, user, VerbType.Alternative)[VerbType.Alternative];
+            var verbs = _verbSystem.GetLocalVerbs(target, user, new() { typeof(AlternativeVerb) });
```
Comment (mirrorcult) [inline Content.Shared/Interaction/SharedInteractionSystem.cs:739]: this is kind of ass syntax as opposed to what it was before, but i think this is uncommon enough that i'm fine with it

**Review by mirrorcult — APPROVED** [review]

File: Content.Shared/Verbs/Verb.cs (hunk 1 above)
Comment (ElectroJr) [reply Content.Shared/Verbs/Verb.cs]: Yeah I should have done something like that. Now using `Type.Name`.

File: Content.Shared/Interaction/SharedInteractionSystem.cs (hunk 2 above)
Comment (ElectroJr) [reply Content.Shared/Interaction/SharedInteractionSystem.cs]: I added a wrapper function for single-type requests, so now its just
```cs
_verbSystem.GetLocalVerbs(target, user, typeof(AlternativeVerb))
```
Which makes it somewhat better.

Comment (mirrorcult) [reply Content.Shared/Interaction/SharedInteractionSystem.cs]: sgtm

**Pull request merged**